src/app/Http/Controllers/ApiController.php

```diff
@@ -12,7 +12,10 @@
 class ApiController extends Controller
 {
     public function locations(Request $request) {
+        $default_page_size = env('LOCATION_PAGE_SIZE', 100);
```

Comment:
I like how you set a variable rather than packing it all into one line like I did. 

This `input` method takes a second argument that is the default to use if no value was sent with the request. Using that argument can eliminate the need for the block of code below. 

`$request->input('page_size', $default_page_size)`

I'm not sure we need to set a max value, but you could with the min function. That's the one thing we'd be missing if the block below is eliminate. 

You can take our leave these style suggestions. We're close enough here.

Reply:
I still think that we need the extra check. 

I agree that using the default value to the `input` method will solve the problem of what happens when this endpoint is called without a `page_size` parameter. 

However, there is still the problem of the user giving a `page_size` value that does not nicely coerce to an integer. In that case, `$default_page_size` is not used but `intval()` will return 0 (which the documentation says is the result of `intval()` for non-number-looking inputs like "testing"). In addition, the requester could circumvent even that by calling this endpoint with a negative `page_limit` which renders impossible the ability to use `min()`. 

I'm just thinking out loud. Do you agree or disagree?

Reply:
I think we're down to coding style changes at this point. I'm good with your thought process on this, which is more thorough than mine. Let's get this out there!